tests/std/tests/P0896R4_ranges_iterator_machinery/test.cpp

```diff
@@ -306,18 +308,21 @@ struct destructible_archetype<0> {
 inline constexpr std::size_t destructible_archetype_max = 1;
 
 // clang-format off
-#define SEMIREGULAR_OPS(prefix)                                                 \
-    prefix##_archetype() requires (I != 1);                                     \
-    prefix##_archetype(prefix##_archetype const&) requires (I != 2);            \
-    prefix##_archetype(prefix##_archetype&&) requires (I == 3) = delete;        \
+#define COPYABLE_OPS(prefix)                                                    \
+    prefix##_archetype(prefix##_archetype const&) requires (I != 1);            \
+    prefix##_archetype(prefix##_archetype&&) requires (I == 2) = delete;        \
                                                                                 \
-    prefix##_archetype& operator=(prefix##_archetype const&) requires (I != 4); \
-    prefix##_archetype& operator=(prefix##_archetype&&) requires (I == 5) = delete
+    prefix##_archetype& operator=(prefix##_archetype const&) requires (I != 3); \
+    prefix##_archetype& operator=(prefix##_archetype&&) requires (I == 4) = delete
+
+#define SEMIREGULAR_OPS(prefix, default_disable)                                \
+    prefix##_archetype() requires (I != default_disable);                       \
+    COPYABLE_OPS(prefix)
 // clang-format on
 
 template <std::size_t I>
 struct semiregular_archetype : destructible_archetype<I> {
-    SEMIREGULAR_OPS(semiregular);
+    SEMIREGULAR_OPS(semiregular, 5);

```

Comment:
it would be slightly cleaner to pass in the whole requires expression, or the subexpression in the ()s to the macro, that way the macro would be more hygenic (since it wouldn't need to access identifiers that aren't associated with it's parameters.

Reply:
it's not a blocker though.

Reply:
actually .... uhhh, is this really saving you any typing anymore at all?

Reply:
`COPYABLE_OPS` is still valuable, but you're right - `SEMIREGULAR_OPS` is more trouble than it's worth now.